core/state/statedb.go

```diff
@@ -633,6 +633,15 @@ func (db *DB) Copy() *DB {
 		}
 		state.stateObjectsDirty[addr] = struct{}{}
 	}
+	for addr := range db.stateValidators {
```

Comment:
You may take a look at the Deep copy method already implemented in the test code: `staking/types/test/copy.go:12`

Reply:
Ah, I must have missed this when I was initially looking at the code. Is there a reason not to import and use the function you mention? All libraries staking/types/test/ uses are already being imported in statedb.go

Reply:
You can definitely reuse the existing code. Just double check the code and make sure there is enough unit tests to cover the new code change.